torch_geometric/loader/imbalanced_sampler.py

```diff
@@ -60,6 +72,10 @@ def __init__(
             assert dataset.num_nodes == y.numel()
             y = y[input_nodes] if input_nodes is not None else y
 
+        elif isinstance(dataset, Tensor):
+            y = dataset.view(-1)
+            y = y[input_nodes] if input_nodes is not None else y
```

Comment:
I find this condition a bit confusing - why would someone want to provide both?

Reply:
I'm also a bit confused about this. Indeed, is it really necessary to support so many types of input for this variable `dataset`? I would suggest to support only `torch.Tensor` (labels) or `List[torch.Tensor]` ( a list of labels) as input.

```python
class ImbalancedSampler(torch.utils.data.WeightedRandomSampler):
    def __init__(
        self,
        y: Union[Tensor, List[Tensor]],
        num_samples: Optional[int] = None,
    ):

        if isinstance(y, Tensor):
            y = y.view(-1)
        else:
            if isinstance(y[0], Tensor):
                y = torch.cat(y, dim=0).view(-1)
            else:
                y = torch.tensor(y).view(-1)

    assert y.dtype == torch.long  # Require classification.

    num_samples = y.numel() if num_samples is None else num_samples

    class_weight = 1. / y.bincount()
    weight = class_weight[y]

    return super().__init__(weight, num_samples, replacement=True)
```
Users need to specify themselves which labels should be inputted
```python
from torch_geometric.loader import ImbalancedSampler

sampler = ImbalancedSampler(data.y[data.train_mask]) 
```

This may result in some changes for relevant users. I'm not so sure. Could you give me some suggestions?

Reply:
Yeah, seems fairly heavily used so maybe best not to break the current contract. You could define a new class for  this approach... probably need @rusty1s for a better opinion on this

Reply:
I personally would like the change of only inputting a `Tensor` which results in s cleaner interface but has the trade-off that users need to first collect their label tensor. I think we can leave the interface as it is for now, and add a deprecation warning for other inputs than tensors. We would also need to update examples. Happy to also do it in a follow-up PR if this is too complicated right now.